memgpt/memory.py

```diff
@@ -16,96 +18,214 @@
     validate_date_format,
 )
 
-# from llama_index import Document
-# from llama_index.node_parser import SimpleNodeParser
 
+class MemoryModule(BaseModel):
+    """Base class for memory modules"""
 
-class CoreMemory(object):
-    """Held in-context inside the system message
+    description: Optional[str] = None
+    limit: int = 2000
+    value: Optional[Union[List[str], str]] = None
 
-    Core Memory: Refers to the system block, which provides essential, foundational context to the AI.
-    This includes the persona information, essential user details,
-    and any other baseline data you deem necessary for the AI's basic functioning.
-    """
+    def __setattr__(self, name, value):
+        """Run validation if self.value is updated"""
+        super().__setattr__(name, value)
+        if name == "value":
+            # run validation
+            self.__class__.validate(self.dict(exclude_unset=True))
 
-    def __init__(self, persona=None, human=None, persona_char_limit=None, human_char_limit=None, archival_memory_exists=True):
-        self.persona = persona
-        self.human = human
-        self.persona_char_limit = persona_char_limit
-        self.human_char_limit = human_char_limit
+    @validator("value", always=True)

```

Comment:
If this is supposed to be defaulting to the class limit it should only be set there.